tests/test_s3_to_json_s3.py

```diff
@@ -0,0 +1,477 @@
+                },
+                "jsonSchema" : {
+                  "type" : "string",
+                  "description" : "The uri for the json schema if the content type is 'application/json'.",
+                  "format" : "uri"
+                },
+                "metadata" : {
+                  "description" : "Any additional metadata about this file."
+                }
+              },
+              "required" : [
+                "filename",
+                "timestamp"
+              ],
+              "additionalProperties" : False,
+              "examples" : [
+                {
+                  "filename" : "foo.json",
+                  "timestamp" : "2022-06-14T11:29:59.915-07:00",
+                  "contentType" : "application/json",
+                  "identifier" : "foo",
+                  "stepPath" : "Bar/foo",
+                  "jsonSchema" : "http://example.org/schemas/v1/Foo.json",
+                  "metadata" : {
+                    "value" : 1
+                  }
+                }
+              ]
+            }
+          },
+          "properties" : {
+            "appName" : {
+              "type" : "string",
+              "description" : "Name of the app that built the archive."
+            },
+            "appVersion" : {
+              "type" : "string",
+              "description" : "Version of the app that built the archive."
+            },
+            "deviceInfo" : {
+              "type" : "string",
+              "description" : "Information about the specific device."
+            },
+            "deviceTypeIdentifier" : {
+              "type" : "string",
+              "description" : "Specific model identifier of the device."
+            },
+            "files" : {
+              "type" : "array",
+              "description" : "A list of the files included in this archive.",
+              "items" : {
+                "$ref" : "#/definitions/FileInfo"
+              }
+            }
+          },
+          "required" : [
+            "appName",
+            "appVersion",
+            "deviceInfo",
+            "deviceTypeIdentifier",
+            "files"
+          ]
+        }
+        return metadata_json_schema
+
+    def test_get_dataset_mapping(self, dataset_mapping_uri):
+        data_mapping = get_data_mapping(data_mapping_uri=dataset_mapping_uri)
+        assert isinstance(data_mapping, dict)
+
+    def test_get_schema_mapping(self, schema_mapping_uri):
+        schema_mapping = get_data_mapping(data_mapping_uri=schema_mapping_uri)
+        assert isinstance(schema_mapping, dict)
+
+    def test_get_archive(self, artifact_bucket, namespace):
+        archive_map = get_archive_map(archive_map_version="v4.4.1")
+        assert isinstance(archive_map, dict)
+
+    def test_update_sts_tokens(self):
+        synapse_data_folder_1 = "syn11111111"
+        sts_tokens = update_sts_tokens(
+                syn=MockSynapse(),
```

Comment:
This is one way to do it.  But you can actually use the `patch.Mock()` class for stuff like this.  or `patch.mock`.

Reply:
Nice! I didn't know about mock.